### ISSUE TITLE
Upgrading to Xcode 14.3 will not build when plugin transitive dependencies have a low deployment target: File not found: XcodeDefault.xctoolchain/usr/lib/arc/libarclite_iphoneos.a

### ISSUE THREAD

**Issue by jmagman** [flutter/flutter#124340]
Steps to reproduce 

1. Check out 3.7.10 or be on master to pick up https://github.com/flutter/flutter/issues/123890
2. 
```
$ flutter create test_create
$ cd test_create
```
3. Add a plugin with a lower-than-9 iOS dependency target (I used `flutter_appauth`) as a pubspec dependency.
4.
```
$ flutter build ios
```

Does not build:
```
Error (Xcode): File not found: /Users/m/Applications/Xcode-14-3.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/arc/libarclite_iphoneos.a


Error (Xcode): Linker command failed with exit code 1 (use -v to see invocation)
```

On macOS this is `libarclite_macosx.a`

Customers are reporting updating the pod's `IPHONEOS_DEPLOYMENT_TARGET` version fixes it.  https://github.com/flutter/flutter/issues/123890#issuecomment-1499278832

**Comment by jmagman** [comment]
This seems to only happen for plugins with low deployment target **transitive** pod dependencies 
Fixed by moving this logic:
https://github.com/flutter/flutter/blob/4fc78b9355c9e9913144858bb40328b7370b8ea0/packages/flutter_tools/bin/podhelper.rb#L85-L89

up above the transitive dependency check so it runs for all pods, not just the Flutter pub pods:
https://github.com/flutter/flutter/blob/4fc78b9355c9e9913144858bb40328b7370b8ea0/packages/flutter_tools/bin/podhelper.rb#L66

Same by moving down the macOS logic:
https://github.com/flutter/flutter/blob/4fc78b9355c9e9913144858bb40328b7370b8ea0/packages/flutter_tools/bin/podhelper.rb#L130-L132

**Comment by jmagman** [comment]
This seems to happen with transitive pods with deployment targets < iOS 9 or < macOS 10.11.

**Comment by jmagman** [comment]
Additionally, this has nothing to do with Flutter, I was able to reproduce with just a pod that targets iOS 8 and CocoaPods without Flutter.

1. Make a new Xcode project.  Note I had to switch to Xcode 13 compatible on CocoaPods 1.12.0.
<img width="258" alt="Screenshot 2023-04-06 at 12 56 15 PM" src="https://user-images.githubusercontent.com/682784/230481349-27663bbb-0012-45bd-9fdb-101a502cf07b.png">

2.
```
$ pod init
```
4. In Podfile add `pod 'AppAuth'`
```ruby
target 'TestAuth' do
  use_frameworks!
  pod 'AppAuth'
end
```
5. 
```
$ open TestAuth.xcworkspace
```
6. Run.  Same linker failure:
```
File not found: /Users/m/Applications/Xcode-14-3.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/arc/libarclite_iphonesimulator.a
```

**Comment by supamunkey** [comment]
With that said, is the best course of action to simply downgrade to XCode 14.2?

**Comment by jmagman** [comment]
You can 
1. Downgrade to Xcode 14.2
2. Add a workaround to `ios/Podfile` and `macos/Podfile` that sets the pods to the same version your Flutter app is targeting.  https://stackoverflow.com/questions/75574268/missing-file-libarclite-iphoneos-a-xcode-14-3

<img width="546" alt="Screenshot 2023-04-06 at 1 23 41 PM" src="https://user-images.githubusercontent.com/682784/230486548-e82ad43a-9905-413d-a606-d2a2057da4a6.png">

**Comment by jmagman** [comment]
https://github.com/flutter/flutter/pull/124349 is available on the [master channe](https://github.com/flutter/flutter/wiki/Flutter-build-release-channels#how-to-change-channels)l.  If someone could remove any workarounds they made to their Podfile or Xcode project and try it out and validate it works, that would be very helpful!

**Comment by jmagman** [comment]
I also recommend filing issues with any pods you see hitting this to suggest they updated to a minimum of at least iOS 9 and macOS 10.11.

For example https://github.com/openid/AppAuth-iOS/pull/761

**Comment by creativecreatorormaybenot** [comment]
@jmagman will this be cherry-picked onto the stable channel?

I would love to try it. However, I do not want to use the other changes that have been pushed onto the master channel.

**Comment by creativecreatorormaybenot** [comment]
> I also recommend filing issues with any pods you see hitting this to suggest they updated to a minimum of at least iOS 9 and macOS 10.11.

How do we find out which of the pods is hitting this?

The only error message I see is:

> Error (Xcode): File not found: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/arc/libarclite_iphoneos.a

**Comment by jmagman** [comment]
Requesting hotfix at https://github.com/flutter/flutter/issues/124529.

**Comment by jmagman** [comment]
This fix is available in Flutter 3.7.11.

**Comment by creativecreatorormaybenot** [comment]
Thanks so much for the quick fix @jmagman 🙏
